core/classifiers/CodeClassifier/CodeClassifierTest.py

```diff
@@ -0,0 +1,101 @@
+
+import json
+import os
+
+from core.classifiers.CodeClassifier import CodeClassifier
+from core.tests import test_utils
+import vmconf
+
+
+def _load_training_data():
+    file_path = os.path.join(vmconf.DATASETS_DIR, 'CodeClassifier.json')
+    with open(file_path, 'r') as f:
+        training_data = json.loads(f.read())
+    return training_data
+
+
+class CodeClassifierTests(test_utils.GenericTestBase):
+    """Tests for code classifier and related preprocessing functions."""
+
+    def setUp(self):
+        super(CodeClassifierTests, self).setUp()
+        # Example programs for testing preprocessing functions.
+        self.program_a = (
+            '# In Python, the code\n#\n#     for letter in [\'a\', \'b\']:\n#'
+            '         print letter\n#\n# prints:\n#\n#     a\n#     '
+            'b\nsum = 0\nfor num in range(1000):\n  if num%3==0 and num%5==0:'
+            '\n\tsum = sum + num\n  else:\n\tif num%3 == 0:\n\t  '
+            'sum = sum + num\n\tif num%5 == 0:\n\t  sum = sum + num\nprint sum')
+
+        self.program_b = (
+            '# In Python, the code\n#\n#     for letter in [\'a\', \'b\']:\n#'
+            '         print letter\n#\n# prints:\n#\n#     a\n#     b\n\n'
+            'for num in range(1000):\n  if num % 3 == 0 or num % 5 == 0:\n'
+            '    sum += num\nprint sum')
+
+        self.data = {
+            1: {
+                'source': self.program_a,
+                'class': 1
+            },
+            2: {
+                'source': self.program_b,
+                'class': 2
+            }
+        }
+
+        self.clf = CodeClassifier.CodeClassifier()
+
+    def test_that_cv_tokenizer_works(self):
+        """Make sure that custom tokenizer used for CountVectorizer is
+        working as expected."""
+        tokens = CodeClassifier.cv_tokenizer(self.program_a)
+        expected_tokens = [
+            'V', '=', '0', 'for', 'V', 'in', 'V', '(', '1000', ')', ':', 'if',
+            'V', '%', '3', '==', '0', 'and', 'V', '%', '5', '==', '0', ':', 'V',
+            '=', 'V', '+', 'V', 'else', ':', 'if', 'V', '%', '3', '==', '0',
+            ':', 'V', '=', 'V', '+', 'V', 'if', 'V', '%', '5', '==', '0', ':',
+            'V', '=', 'V', '+', 'V', 'print', 'V']
+
+        self.assertListEqual(tokens, expected_tokens)
+
+    def test_that_token_to_id_is_correct(self):
+        """Make sure that correct token_to_id map is generated."""
+        token_to_id = CodeClassifier.map_tokens_to_ids(self.data, 0)
+        expected_tokens = [
+            'and', 'UNK', '%', 'for', ')', '(', '+', 'V', 'else', '==', '0',
+            '3', '5', '1000', 'in', 'print', ':', '=', 'or', '+=', 'if']
+        self.assertListEqual(token_to_id.keys(), expected_tokens)
+
+    def test_that_jaccard_index_is_calculated_correctly(self):
+        """Make sure that correct jaccard index is calculated between
+        two sets."""
+        set_a = [1, 2, 2, 3]
+        set_b = [2, 3, 4]
+        expected_index = 0.4
```

Comment:
shouldn't it be 2/7=0.286 according to your logic? set_a and set_b have 2 elements in common (2 & 3) while they have 7 common overall. I think I'm missing something here.